diff --git a/js/car.js b/js/car.js
index aa0d2c4..403aa62 100644
--- a/js/car.js
+++ b/js/car.js
@@ -16,8 +16,33 @@ class Car {
 
     update(roadBorders) {
         this.#moment();
+        this.polygon = this.#createPolygon();
         this.sensor.update(roadBorders);
     }
+    #createPolygon() {
+        const points = [];
+        const rad = Math.hypot(this.width, this.height) / 2;
+        const alpha = Math.atan2(this.width, this.height);
+        points.push({
+            x: this.x - Math.sin(this.angle - alpha) * rad,
+            y: this.y - Math.cos(this.angle - alpha) * rad   
+        });
+        points.push({
+            x: this.x - Math.sin(this.angle + alpha) * rad,
+            y: this.y - Math.cos(this.angle + alpha) * rad
+        });
+        points.push({
+            x: this.x - Math.sin(Math.PI + this.angle - alpha) * rad,
+            y: this.y - Math.cos(Math.PI + this.angle - alpha) * rad
+        }); 
+        points.push({
+            x: this.x - Math.sin(Math.PI + this.angle + alpha) * rad,
+            y: this.y - Math.cos(Math.PI + this.angle + alpha) * rad
+        });
+
+        return points;
+    }
+
     #moment() {
         if (this.controls.forward) {
             this.speed += this.acceleration;
